### DISCUSSION TITLE
When using useNumberField, is there any way to get information about the number formatting being used?

### DISCUSSION THREAD

**Discussion by mellis481** [adobe/react-spectrum#4872] [Q&A] [answered]
I'm currently POCing the `useNumberField` hook ([documentation](https://react-spectrum.adobe.com/react-aria/useNumberField.html)) which essentially provides localized number masking. This means:
- for locales like `en-US`, commas are used as thousand separators and a period is use as the decimal separator.
- for locales like `fr-FR`, spaces are used as thousand separators and a comma is use as the decimal separator.

I'm wondering if react-spectrum provides access (a helper function) to the localized number formatting it ends up using in `useNumberField` such that it could be used by consumers to inform users about what characters are allowed.

For instance, if I could tap into a function that would return the following based off the current locale:
```
{
  thousands: 'comma',
  decimalSeparator: 'period'
}
```
... then I could include text like the following to further help the user:  _Your number can be formatted using a **comma** for the thousands separator and a **period** for the decimal separator._

**Comment by reidbarber** [comment, marked as answer]
The [Intl.NumberFormat.formatToParts()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/formatToParts) function may be useful, where you can check the `type` of a part as `"group"` or `"decimal"`.
